dapps/kiosk-cli/index.js

```diff
@@ -0,0 +1,841 @@
+  const kioskArg = txb.object(kioskCap.content.fields.for);
+  const capArg = txb.objectRef({ ...kioskCap });
+  delist(txb, item.data.type, kioskArg, capArg, itemId);
+
+  return sendTx(txb);
+}
+
+/**
+ * Command: `purchase`
+ * Description: Purchases an item from the specified Kiosk
+ *
+ * TODO:
+ * - add destination "kiosk" or "user" (kiosk by default)
+ */
+async function purchaseItem(itemId, opts) {
+  const { target, kiosk: inputKioskId } = opts;
+
+  if (target && target !== 'kiosk' && !isValidSuiAddress(target)) {
+    throw new Error(
+      'Invalid target address: "%s"; use "kiosk" if you want to store in your Kiosk',
+      target,
+    );
+  }
+
+  if (inputKioskId && !isValidSuiObjectId(inputKioskId)) {
+    throw new Error('Invalid Kiosk ID: "%s"', inputKioskId);
+  }
+
+  if (!isValidSuiObjectId(itemId)) {
+    throw new Error('Invalid Item ID: "%s"', itemId);
+  }
+
+  let kioskId = inputKioskId;
+
+  const itemInfo = await provider.getObject({
+    id: itemId,
+    options: { showType: true, showOwner: true },
+  });
+
+  if ('error' in itemInfo || !itemInfo.data) {
+    throw new Error(`Item ${itemId} not found; ${itemInfo.error}`);
+  }
+
+  if (!('ObjectOwner' in itemInfo.data.owner)) {
+    throw new Error(`Item ${itemId} is not owned by an object`);
+  }
+
+  if (!kioskId) {
+    const itemKeyId = itemInfo.data.owner.ObjectOwner;
+    const itemKey = await provider.getObject({ id: itemKeyId, options: { showOwner: true } });
+
+    if ('error' in itemKey || !itemKey.data) {
+      throw new Error(`Dynamic Field ${itemId} key not found; ${itemKey.error}`);
+    }
+
+    if (!('ObjectOwner' in itemKey.data.owner)) {
+      throw new Error(`Dynamic Field ${itemId} key is not owned by an object`);
+    }
+
+    kioskId = itemKey.data.owner.ObjectOwner;
+  }
+
+  const [kiosk, policies, listing] = await Promise.all([
+    provider.getObject({ id: kioskId, options: { showOwner: true } }),
+    queryTransferPolicy(provider, itemInfo.data.type),
+    provider.getDynamicFieldObject({
+      parentId: kioskId,
+      name: { type: '0x2::kiosk::Listing', value: { id: itemId, is_exclusive: false } },
+    }),
+  ]);
+
+  if ('error' in listing || !listing.data) {
+    throw new Error(`Item ${itemId} not listed in Kiosk ${kioskId}`);
+  }
+
+  if ('error' in kiosk || !kiosk.data) {
+    throw new Error(`Kiosk ${kioskId} not found`);
+  }
+
+  if ('error' in itemInfo || !itemInfo.data) {
+    throw new Error(`Item ${itemId} not found`);
+  }
+
+  if (policies.length === 0) {
+    throw new Error(`No transfer policy found for type ${itemInfo.data.type}`);
+  }
+
+  const price = listing.data.content.fields.value;
+  const txb = new TransactionBlock();
+  const item = purchaseAndResolvePolicies(
+    txb,
+    itemInfo.data.type,
+    { price },
+    kioskId,
+    itemInfo.data.objectId,
+    policies[0],
+  );
+
+  // For the locking policy scenario when an item needs to be locked;
+  if (item === null) {
```

Comment:
This has changed on 0.3.0 version.
`const { item, canTransfer } = purchaseAndResolvePolicies()`